druid-shell/src/backend/gtk/window.rs

```diff
@@ -1021,13 +1021,27 @@ impl WindowHandle {
             let scale = state.scale.get();
             let (width_px, height_px) = state.window.size();
             let alloc_px = state.drawing_area.allocation();
-            let window = Size::new(width_px as f64, height_px as f64).to_dp(scale);
-            let alloc = Rect::from_origin_size(
-                (alloc_px.x as f64, alloc_px.y as f64),
-                (alloc_px.width as f64, alloc_px.height as f64),
-            )
-            .to_dp(scale);
-            window.to_rect() - alloc
+
+            if let Some(window) = state.window.window() {
+                let frame = window.frame_extents();
+                let (pos_x, pos_y) = window.position();
+                Insets::new(
+                    (pos_x - frame.x) as f64,
+                    (pos_y - frame.y) as f64,
+                    (frame.x + frame.width - (pos_x + width_px)) as f64,
+                    (frame.y + frame.height - (pos_y + height_px)) as f64,
+                )
+                .to_dp(scale)
+                .nonnegative()
```

Comment:
Does this take into account the menu bar? We want the insets to give the location of the drawing area: the menu should be excluded, just like the window decorations are.

Reply:
> Does this take into account the menu bar? 

Sorry, just pushed a fix for that.

Reply:
Don't forget to delete the obsolete doc comment!